internal/backend/config/set.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"fmt"
+
+	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
+)
+
+type ComponentConfig interface {
+	Validate() error
+}
+
+type emptyComponentConfig struct{}
+
+func (emptyComponentConfig) Validate() error { return nil }
+
+var EmptyComponentConfig emptyComponentConfig
+
+type baseSet interface {
+	hasDefault() bool
+	default_() ComponentConfig
```

Comment:
Not happy with `default_` and in general lower case, unexported, functions in an interface.

Reply:
removed all functions, now only constraining known sets.